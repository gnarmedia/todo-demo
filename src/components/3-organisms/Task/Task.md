```jsx
function handleToggleStatusClick(id) {
  alert(`toggle status: ${id}`);
  return false;
}
function handleRemoveClick(id) {
  alert(`task removed: ${id}`);
  return false;
}
<Task
  title="Sample Task"
  isComplete={false}
  onToggleStatusClick={() => handleToggleStatusClick(1)}
  onRemoveClick={() => handleRemoveClick(1)}
/>
```