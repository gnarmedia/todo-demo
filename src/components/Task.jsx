// import dependencies
import React, { Fragment } from "react";
import PropTypes from "prop-types";

/**
 * @name: Task
 * @desc: A task with status
 * @param {object} task
 * @returns: {function}
 */
const Task = ({ title, isComplete, onToggleStatusClick }) => {
    return (
        // Fragments are used to wrap multiple rendered elements within a component
        <Fragment>
            {/* sets the check based on status */}
            <input
                type="checkbox"
                defaultChecked={isComplete}
                onClick={onToggleStatusClick}
            />
            {title}&nbsp;[{isComplete ? "complete" : "incomplete"}]
        </Fragment>
    );
};

Task.propTypes = {
    title: PropTypes.string.isRequired,
    isComplete: PropTypes.bool.isRequired,
    onToggleStatusClick: PropTypes.func.isRequired
};

// export our component for importing into other modules
export default Task;
