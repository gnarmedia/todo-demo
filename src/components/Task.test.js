import React from "react";
import { shallow } from "enzyme";

// import components
import Task from "./Task.jsx";

// load some fake tasks
const task = {
        id: 1,
        title: "Task 1",
        isComplete: false
    },
    onToggleStatusClickMock = jest.fn();

function _renderTask({
    title = "Task",
    isComplete = false,
    onToggleStatusClick = () => {}
}) {
    return (
        <Task
            title={title}
            isComplete={isComplete}
            onToggleStatusClick={onToggleStatusClick}
        />
    );
}

describe("Task", function() {
    it("should render the Task with completed status", () => {
        const tree = shallow(
            _renderTask({ title: "Completed Task", isComplete: true })
        );

        expect(tree).toMatchSnapshot();
    });

    it("should render the Task with incomplete status", () => {
        const tree = shallow(_renderTask({ title: "Incomplete Task" }));

        expect(tree).toMatchSnapshot();
    });

    it("should call onToggleStatusClick when checkbox is clicked", () => {
        const onToggleStatusClickMock = jest.fn(),
            wrapper = shallow(
                _renderTask({ onToggleStatusClick: onToggleStatusClickMock })
            );

        expect(wrapper.find("input").exists()).toEqual(true);
        wrapper.find("input").simulate("click");
        expect(onToggleStatusClickMock).toHaveBeenCalled();
    });
});
