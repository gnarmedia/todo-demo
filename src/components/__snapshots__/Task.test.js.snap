// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render the Task with completed status 1`] = `
<React.Fragment>
  <input
    defaultChecked={true}
    onClick={[Function]}
    type="checkbox"
  />
  Completed Task
   [
  complete
  ]
</React.Fragment>
`;

exports[`Task should render the Task with incomplete status 1`] = `
<React.Fragment>
  <input
    defaultChecked={false}
    onClick={[Function]}
    type="checkbox"
  />
  Incomplete Task
   [
  incomplete
  ]
</React.Fragment>
`;
